sherpa-ncnn/csrc/recognizer.cc

```diff
@@ -124,23 +128,77 @@ class Recognizer::Impl {
     return stream;
   }
 
+  std::unique_ptr<Stream> CreateStream(const char* contexts) const {
```

Comment:
Could you add a member to `RecognizerConfig`
```
std::string contexts;
```
and remove this method `CreateStream(const char* contexts)`?

If `RecognizerConfig.contexts` is not empty, we load it when creating the recognizer.
When creating the stream, if `RecognizerConfig.contexts` is not empty, we create a stream that supports hotword detection. Otherwise, we just create a normal stream.

Reply:
done，btw, add c-api to support  hotwords,  using bin/decode-file-c-api to test.